src/ast/mod.rs

```diff
@@ -3857,6 +3857,31 @@ pub enum Statement {
         managed_location: Option<String>,
     },
     /// ```sql
+    /// CREATE [ OR REPLACE ] [ TRANSIENT ] DATABASE [ IF NOT EXISTS ] <name>
+    ///     [ CLONE <source_schema>
+    ///         [ { AT | BEFORE } ( { TIMESTAMP => <timestamp> | OFFSET => <time_difference> | STATEMENT => <id> } ) ]
+    ///         [ IGNORE TABLES WITH INSUFFICIENT DATA RETENTION ]
+    ///         [ IGNORE HYBRID TABLES ] ]
+    ///     [ DATA_RETENTION_TIME_IN_DAYS = <integer> ]
+    ///     [ MAX_DATA_EXTENSION_TIME_IN_DAYS = <integer> ]
+    ///     [ EXTERNAL_VOLUME = <external_volume_name> ]
+    ///     [ CATALOG = <catalog_integration_name> ]
+    ///     [ REPLACE_INVALID_CHARACTERS = { TRUE | FALSE } ]
+    ///     [ DEFAULT_DDL_COLLATION = '<collation_specification>' ]
+    ///     [ STORAGE_SERIALIZATION_POLICY = { COMPATIBLE | OPTIMIZED } ]
+    ///     [ COMMENT = '<string_literal>' ]
+    ///     [ CATALOG_SYNC = '<snowflake_open_catalog_integration_name>' ]
+    ///     [ CATALOG_SYNC_NAMESPACE_MODE = { NEST | FLATTEN } ]
+    ///     [ CATALOG_SYNC_NAMESPACE_FLATTEN_DELIMITER = '<string_literal>' ]
+    ///     [ [ WITH ] TAG ( <tag_name> = '<tag_value>' [ , <tag_name> = '<tag_value>' , ... ] ) ]
+    ///     [ WITH CONTACT ( <purpose> = <contact_name> [ , <purpose> = <contact_name> ... ] ) ]
```

Comment:
Re documentation, i think using a simple example or only mentioning that it represents the `CREATE DATABASE` statement would suffice, the actual spec we can delegate to the linked documentation for each dialect.